cryptography/src/bls12381/dkg.rs

```diff
@@ -328,62 +329,17 @@ pub enum Error {
     DkgFailed,
 }
 
-/// Recover public polynomial by interpolating coefficient-wise all
-/// polynomials using precomputed Barycentric Weights.
-///
-/// It is assumed that the required number of commitments are provided.
-fn recover_public_with_weights<V: Variant>(
-    commitments: &BTreeMap<u32, poly::Public<V>>,
-    weights: &BTreeMap<u32, poly::Weight>,
-    threshold: u32,
-    concurrency: usize,
-) -> poly::Public<V> {
-    let work = |coeff| {
-        // Extract evaluations for this coefficient from all commitments
-        let evals = commitments
-            .iter()
-            .map(|(dealer, commitment)| poly::Eval {
-                index: *dealer,
-                value: commitment.get(coeff),
-            })
-            .collect::<Vec<_>>();
-
-        // Use precomputed weights for interpolation
-        msm_interpolate(weights, &evals).expect("interpolation should not fail")
-    };
-    let range = 0..threshold;
-    if concurrency <= 1 || threshold <= 1 {
-        range.map(work).collect()
-    } else {
-        // Build a thread pool with the specified concurrency
-        let pool = ThreadPoolBuilder::new()
-            .num_threads(concurrency)
-            .build()
-            .expect("Unable to build thread pool");
-
-        // Recover signatures
-        pool.install(move || {
-            range
-                .into_par_iter()
-                .map(work)
-                .collect::<Vec<_>>()
-                .into_iter()
-                .collect()
-        })
-    }
-}
-
 /// The output of a successful DKG.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Output<V: Variant, P> {
     summary: Summary,
     players: Set<P>,
-    public: Public<V>,
+    public: Sharing<V>,
```

Comment:
Making `Mode` now makes sense to me (as something you'd persist with the sharing to look at some historical output)